### PR TITLE
Add workflow publishing ecdsa contracts for dev env 

### PR DESCRIPTION
**Pull request by michalinacienciala**
We're adding a workflow which will publish a package with compiled
`ecdsa` contracts. Package will be meand for development purposes.

The `-dev` suffix will be used in the name of the published package.
The workflow will be triggered every time a change to contracts or
dependencies gets pushed to `main`. Also a change to the workflow file
will trigger the workflow.

As art of this PR we're also adding ignoring of `external` directory in git and linters.

TODO:
Once PR gets "pre-approved":
- [x] Remove `--dry-run`
- [x] Publish first NPM package (step not needed, package was published manually)
- [x] Remove `npm-ecdsa` branch trigger
- [x] Uncomment "Bump up package version" step

### CODE REVIEW

**Comment by michalinacienciala** [comment]
@Kuba, please see if PR looks ok to you. Once I get a green light from you, I will perform steps from the TODO list and ask you for final PR approval.

File: .github/workflows/npm-ecdsa.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: NPM ECDSA
+
+on:
+  push:
+    branches:
+      - main
+      - npm-ecdsa # TODO: remove before merging to main
+    paths:
+      - "solidity/ecdsa/contracts**"
```
Comment (nkuba) [inline .github/workflows/npm-ecdsa.yml:9]: Shouldn't it be `contracts/**`?
```suggestion
      - "solidity/ecdsa/contracts/**"
```

Comment (nkuba) [inline .github/workflows/npm-ecdsa.yml:9]: Let's also add `"solidity/ecdsa/deploy/**"` directory as we may use the deployment scripts in other projects.

File: .github/workflows/npm-ecdsa.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      #     work-dir: ./solidity/ecdsa
+      #     environment: dev
+      #     branch: ${{ github.ref }}
+      #     commit: ${{ github.sha }}
+
+      # TODO: Remove --dry-run before merging to main
+      - name: Publish package
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: npm publish --access=public --network=development --tag=development --dry-run
```
Comment (nkuba) [inline .github/workflows/npm-ecdsa.yml:55]: Looks like we don't use the `network` argument in the publication scripts in this project.

```suggestion
        run: npm publish --access=public --tag=development --dry-run
```

File: solidity/ecdsa/.gitignore [hunk 3]
```diff
@@ -2,5 +2,6 @@
 build/
 cache/
 export/
+external/
```
Comment (nkuba) [inline solidity/ecdsa/.gitignore:5]: Let's exclude `external/npm` directory as it contains the auto-generated content. We may want to add e.g. `external/mainnet` directory with hardcoded mainnet contracts to use.

```suggestion
external/npm
```

File: solidity/ecdsa/package.json [hunk 4]
```diff
@@ -3,6 +3,16 @@
...
   "description": "Keep ECDSA Wallet",
   "license": "MIT",
+  "files": [
+    "artifacts/",
+    "build/contracts/",
+    "contracts/",
+    "!**/test/",
+    "deploy/",
+    "export/",
+    "scripts/",
```
Comment (nkuba) [inline solidity/ecdsa/package.json:13]: We don't have `scripts` directory in this project.
```suggestion
```

**Review by nkuba — CHANGES_REQUESTED** [review]

**Comment by nkuba** [comment]
We need https://github.com/keep-network/keep-core/pull/2874 to get in before this one.

**Review by nkuba — DISMISSED** [review]

**Review by nkuba — APPROVED** [review]

**Pull request merged**